clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -523,6 +523,12 @@ void CodeGenFunction::FinishFunction(SourceLocation EndLoc) {
     NormalCleanupDest = Address::invalid();
   }
 
+  if (getLangOpts().Coroutines && isCoroutine()) {
+    auto *Record = FnRetTy->getAsCXXRecordDecl();
+    if (Record && Record->hasAttr<CoroAwaitElidableAttr>())
+      CurFn->addFnAttr(llvm::Attribute::CoroGenNoallocRamp);
+  }
```

Comment:
I don't like this. It will add this attribute to all the coroutines with the attribute `[[[clang::coro_await_elidable]]](https://github.com/llvm/llvm-project/pull/99282/files#top)`. In practice it may always cause the CoroSplit pass to generate additional codes.

What I thought is, in CoroSplit pass, we can visit every user of the function, and if any user of the function is a call with the elide attribute, we can generate .noalloc variant for the current coroutine.

Reply:
Original intention was to not introduce extra work (for loop over usages) for non attributed code. The check won't have a quick exit path for unattributed code.

Reply:
I will apply this suggestion in the next push.